### ISSUE TITLE
"unreachable code: ICE: EmbeddedCurveAdd expects two register arguments and one array result" in <brillig>

### ISSUE THREAD

**Issue by skaunov** [noir-lang/noir#4687]
### Aim

run some tests for my `lib`

### Expected Behavior

usual `nargo test` report

### Bug

```console
[plume] Running 6 test functions
The application panicked (crashed).
Message:  internal error: entered unreachable code: ICE: EmbeddedCurveAdd expects two register arguments and one array result
Location: compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_black_box.rs:316
```

### To Reproduce

1. just run `nargo test`

### Project Impact

Blocker

### Impact Context

I suspect the code I have hits some internal edge case or something. Too little info to say more precisely.

### Workaround

None

### Installation Method

Binary (`noirup` default)

### Nargo Version

nargo version = 0.26.0 noirc version = 0.26.0+a0f7474ae6bd74132efdb945d2eb2383f3913cce

### Would you like to submit a PR for this Issue?

Maybe

### Support Needs

Teach me `brillig`. X))

**Comment by vezenovm** [comment]
> 1. just run `nargo test`

Could you post the code we should run this on?

**Comment by skaunov** [comment]
It's too dirty for me to publish it on GH. =( I can upload it to a repo 
and add access for you. :shrug:

Would be happy to have a chance to get to cleaning it, btw. =(

**Comment by vezenovm** [comment]
> Would be happy to have a chance to get to cleaning it, btw. =(

It would be very helpful if you could provide a small reproduction as it is hard to test any fixes otherwise

**Comment by skaunov** [comment]
:100: 
The issue is created on @zpedro suggestion, and I literally did it with this comment.
> I'd like to produce a more helpful one; locate where it happens, make an MRE. Usual stuff, you know...

It's kind of pile of helpers which I personally think could be useful libs on they own. But I didn't meet interest in EC arithmetic over `BigInt` or a `hash_to_curve`. So it's still waiting until I could get to it properly.

**Comment by TomAFrench** [comment]
This is closed by #4729
